### PR TITLE
[HW8] [js_hw6 task_02]

### PR DESCRIPTION
**Pull request by Maximus2806**

### CODE REVIEW

File: js_hw6/task_02.js [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+     for (const obj in enterprises) {
+        console.log(`${enterprises[obj].name} (${getAllEmployeesInEnterprise(enterprises[obj].departments)} сотрудников)`);
+        enterprises[obj].departments.forEach((department) => {
+            console.log(`- ${department.name} (${(department.employees_count) === 0 ? 'нет' : department.employees_count} сотрудников)`);
+        });
+     };
+    };
+
+
+function getAllEmployeesInEnterprise (departments) {
```
Comment (VladimirLQA) [inline js_hw6/task_02.js:87]:
```suggestion
function getAllEmployeesInEnterprise (departments) {
if (!departments.length) return 'нет'
```

File: js_hw6/task_02.js [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+// - Отдел разработки (50 сотрудников)
+// - Отдел маркетинга (20 сотрудников)
+// - Отдел охраны труда (5 сотрудников)
+// Предприятие 3 (нет сотрудников)
+// - Отдел аналитики (нет сотрудников)
+
+function logEmployess (enterprises) {
+     for (const obj in enterprises) {
+        console.log(`${enterprises[obj].name} (${getAllEmployeesInEnterprise(enterprises[obj].departments)} сотрудников)`);
+        enterprises[obj].departments.forEach((department) => {
```
Comment (VladimirLQA) [inline js_hw6/task_02.js:80]:
```suggestion
       if (enterprises[obj].departments.length)
        enterprises[obj].departments.forEach((department) => {
```

File: js_hw6/task_02.js [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+// 2. Написать функцию, которая будет принимать 1 аргумент (id отдела или название отдела и возвращать название предприятия, к которому относится).
+// Пример:
+// getEnterpriseName(4) // Предприятие 1
+// getEnterpriseName("Отдел маркетинга") // Предприятие 2
+function getEntrepriseName (depIdOrName) {
+    const enterpriseName = getDepartment(depIdOrName);
+    return enterpriseName.name;        
+    };
+    
+function getDepartment (depIdOrName) {
```
Comment (VladimirLQA) [inline js_hw6/task_02.js]: `getDepartment` - but return value is enterprise !

File: js_hw6/task_02.js [hunk 4]
```diff
@@ -0,0 +1,267 @@
...
+    return (countAll === 0) ? 'нет' : countAll;
+};
+
+logEmployess(enterprises);
+
+// 2. Написать функцию, которая будет принимать 1 аргумент (id отдела или название отдела и возвращать название предприятия, к которому относится).
+// Пример:
+// getEnterpriseName(4) // Предприятие 1
+// getEnterpriseName("Отдел маркетинга") // Предприятие 2
+function getEntrepriseName (depIdOrName) {
```
Comment (VladimirLQA) [inline js_hw6/task_02.js:101]: For this task you can implement full logic using only this function, but further you'll need the helper functions

File: js_hw6/task_02.js [hunk 5]
```diff
@@ -0,0 +1,267 @@
...
+    const enterpriseName = getDepartment(depIdOrName);
+    return enterpriseName.name;
```
Comment (VladimirLQA) [inline js_hw6/task_02.js]: Can do like this
```suggestion
    const {name } = getDepartment(depIdOrName);
    return name;        
```

File: js_hw6/task_02.js [hunk 6]
```diff
@@ -0,0 +1,267 @@
...
+function deleteDepartment(id) {
+    const department = findDepartment(id);    
+    if (department.employees_count === 0) throw new Error (`Department with id ${id} has no employees`);
+    for (const enterprise of enterprises) {
+        for (const department of enterprise.departments) {
+            if (department.id === id && department.employees_count > 0) {
+                enterprise.departments.splice(enterprise.departments.indexOf(department), 1);
+            }
+        }
+    }
```
Comment (VladimirLQA) [inline js_hw6/task_02.js]: You already checked whether the department has an employee
 ```suggestion
                let index = enterprise.departments.findIndex((el) => department.id === el.id);
                enterprise.departments.splice(index, 1);
    }
```

File: js_hw6/task_02.js [hunk 7]
```diff
@@ -0,0 +1,267 @@
...
+
+// Пример:
+// moveEmployees(2, 3)
+
+function moveEmployees(depIdFrom, depIdTo) {
+    if (depIdFrom === depIdTo) throw new Error ('Department ids should not be the same!')
+    const depFrom = findDepartment(depIdFrom);
+    const depTo = findDepartment(depIdTo);
+    if (depFrom.employees_count === 0) throw new Error (`Department with id ${depIdFrom} has no employees`);
+    depTo.employees_count += depFrom.employees_count;
```
Comment (VladimirLQA) [inline js_hw6/task_02.js:259]:
```suggestion
if (depFrom && depTo && getEnterpriseByDeptIDorName(depIdFrom).id === depIdTo(whereToDepartID).id)
    depTo.employees_count += depFrom.employees_count;
```

**Review by VladimirLQA — APPROVED** [review]

File: js_hw6/task_02.js (hunk 3 above)
Comment (Maximus2806) [reply js_hw6/task_02.js]: changed function title to getEnterprise

File: js_hw6/task_02.js (hunk 5 above)
Comment (Maximus2806) [reply js_hw6/task_02.js]: done

File: js_hw6/task_02.js (hunk 6 above)
Comment (Maximus2806) [reply js_hw6/task_02.js]: Good catch, thanks

**Pull request merged**